forte/data/data_store.py

```diff
@@ -0,0 +1,239 @@
+from typing import Union, Type, List, Iterator
+
+import uuid
+from forte.data.ontology.core import EntryType
+from forte.data.base_store import BaseStore
+
+
+class DataStore(BaseStore):
+    def __init__(self):
+        """An implementation from the dataframe-like data store object.
+
+        A collection of NLP entries are stored by types in SortedLists.
+        For example, subtypes of annotations, links, and groups are stored in
+        separate SortedLists.
+        An array records all SortedLists in the format of:
+        [..., Document SortedList, Sentence SortedList, ...].
+        This class records the order of the array and assigns type id,
+        which is the index of each sortedlist, to each entry type.
+
+        Each entry list in the SortedList has the format of
+        [entry_type, tid, begin, end, attr_1, attr_2, ..., attr_n]
+        The first four fields are compulsory for every entry type.
+        Each entry type has a fixed field of attributes.
+        E.g. Document SortedList has lists of structure:
+        [entry_type, tid, begin, end, document_class,sentiment,classifications]
+
+        Args:
+            pack_name (Optional[str], optional): A name for this data store.
+        """
+        super().__init__()
+        self._text = ""
+
+        """
+        The _type_attributes is a private dictionay that provides entry types
```

Comment:
when writing documentation, use this ` to wrap around the variable names, so you should do `_type_attributes` instead of _type_attributes